wemake_python_styleguide/visitors/ast/builtins.py

```diff
@@ -133,34 +136,90 @@ def _check_modulo_patterns(
+        )
+        if not has_f_components:
+            # If no formatted values
+            self.add_violation(
+                complexity.TooComplexFormattedStringViolation(node),
+            )
+            return
+
         for string_component in node.values:
             if isinstance(string_component, ast.FormattedValue):
-                if self._valid_f_value(string_component):
+                # Test if possible chaining is invalid
+                if not self._is_valid_chaining(string_component):
+                    self.add_violation(
+                        complexity.TooComplexFormattedStringViolation(node),
+                    )
+                    return
+                if self._is_valid_formatted_value(string_component.value):
                     continue
                 # Everything else is too complex
                 self.add_violation(
                     complexity.TooComplexFormattedStringViolation(node),
                 )
                 break
 
-    def _valid_f_value(self, node: ast.FormattedValue) -> bool:
-        f_value = node.value
+    def _is_valid_formatted_value(self, f_value: ast.Expr) -> bool:
         # Function call with empty arguments is okay
         if isinstance(f_value, ast.Call) and not f_value.args:
             return True
         # Named lookup, Index lookup & Dict key is okay
         elif isinstance(f_value, ast.Subscript):
-            if self._valid_f_lookup(f_value):
+            if self._is_valid_f_lookup(f_value):
                 return True
         # Variable lookup is okay
         elif isinstance(f_value, ast.Name):
             return True
+        # A single attribute is okay
+        elif isinstance(f_value, ast.Attribute):
+            return True
         return False
 
-    def _valid_f_lookup(self, subscript: ast.Subscript) -> bool:
-        allowed_types = (ast.Str, ast.Num, ast.Name)
+    def _is_valid_chaining(self, node: ast.FormattedValue) -> bool:
+        f_value = node.value
+        parts = attributes.parts(f_value)
+        chained_parts = []
+        
+        try:
+            # Should throw on at latest the fifth attempt
+            for _ in range(5):
+                chained_parts.append(next(parts))
+        except StopIteration:
+            res = True
+            has_invalid_parts = any(
+                not self._is_valid_formatted_value(part)
+                for part in chained_parts
+            )
+            if has_invalid_parts:
+                res = False
+            if len(chained_parts) == 4:
+                # If there are 4 elements, they must have exactly 2 of
+                # subscript or call. This is the alternating case fcn().fcn()
+                n_calls = sum(
+                    isinstance(part, self._allowed_chains)
+                    for part in chained_parts
+                )
+                res = n_calls == 2
+            elif len(chained_parts) == 3:
```

Comment:
This function is going to be reported as too complex.

Reply:
Thank you for the feedback! We are however still struggling with how to rewrite 
```
for _ in range(5):
    chained_parts.append(next(parts))
```
To pass the linter.